### ISSUE TITLE
Slowly get 403s on new install until it doesnt work

### ISSUE THREAD

**Issue by PipeItToDevNull** [omarroth/invidious#813]
When invidious is first stood up on docker I can load the main page, but after interacting with it for a few minutes I can access fewer pages, losing access to `/login` and `/preferences` then to the full site.

**Comment by PipeItToDevNull** [comment]
Docker logs:

```
┌[ssh][groot@geordi.dev0.sh][~]
┕ docker logs invidious_invidious_1 
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Cannot establish connection (PQ::ConnectionError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Error connecting to 'postgres:5432': Connection refused (Errno)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: role "kemal" does not exist (PQ::PQError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: role "kemal" does not exist (PQ::PQError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: role "kemal" does not exist (PQ::PQError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Cannot establish connection (PQ::ConnectionError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Error connecting to 'postgres:5432': Connection refused (Errno)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
```

**Comment by PipeItToDevNull** [comment]
When inside the invidious container I see this:

```
/invidious # cat config/config.yml
channel_threads: 1
feed_threads: 1
db:
  user: kemal
  password: kemal
  host: postgres
  port: 5432
  dbname: invidious
full_refresh: false
https_only: false
domain:
top_enabled: true
captcha_enabled: true
login_enabled: true
registration_enabled: true
/invidious # curl postgres:5432
curl: (52) Empty reply from server
```

**Comment by PipeItToDevNull** [comment]
These are the logs from the postgres container:

```
### first run - setting up invidious database
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data ... ok
creating subdirectories ... ok
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default timezone ... Etc/UTC
selecting dynamic shared memory implementation ... posix
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok

Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/data -l logfile start


WARNING: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
****************************************************
WARNING: No password has been set for the database.
         This will allow anyone with access to the
         Postgres port to access your database. In
         Docker's default configuration, this is
         effectively any other container on the same
         system.

         Use "-e POSTGRES_PASSWORD=password" to set
         it in "docker run".
****************************************************
waiting for server to start....2019-10-09 01:33:43.396 UTC [43] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2019-10-09 01:33:43.420 UTC [44] LOG:  database system was shut down at 2019-10-09 01:33:41 UTC
2019-10-09 01:33:43.425 UTC [43] LOG:  database system is ready to accept connections
 done
server started

/usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*

waiting for server to shut down....2019-10-09 01:33:43.481 UTC [43] LOG:  received fast shutdown request
2019-10-09 01:33:43.492 UTC [43] LOG:  aborting any active transactions
2019-10-09 01:33:43.499 UTC [43] LOG:  worker process: logical replication launcher (PID 50) exited with exit code 1
2019-10-09 01:33:43.500 UTC [45] LOG:  shutting down
2019-10-09 01:33:43.642 UTC [43] LOG:  database system is shut down
 done
server stopped

PostgreSQL init process complete; ready for start up.

2019-10-09 01:33:43.713 UTC [6] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2019-10-09 01:33:43.713 UTC [6] LOG:  listening on IPv6 address "::", port 5432
2019-10-09 01:33:43.765 UTC [6] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2019-10-09 01:33:43.830 UTC [52] LOG:  database system was shut down at 2019-10-09 01:33:43 UTC
2019-10-09 01:33:43.835 UTC [6] LOG:  database system is ready to accept connections
2019-10-09 01:33:43.941 UTC [59] FATAL:  role "kemal" does not exist
2019-10-09 01:33:46.733 UTC [60] FATAL:  role "kemal" does not exist
2019-10-09 01:33:49.801 UTC [62] FATAL:  role "kemal" does not exist
/var/run/postgresql:5432 - accepting connections
### importing table schemas
CREATE ROLE
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
### invidious database setup finished
running postgres /usr/local/bin/docker-entrypoint.sh postgres
2019-10-09 01:33:55.814 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2019-10-09 01:33:55.814 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2019-10-09 01:33:55.912 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2019-10-09 01:33:56.088 UTC [21] LOG:  database system was interrupted; last known up at 2019-10-09 01:33:51 UTC
2019-10-09 01:33:56.344 UTC [21] LOG:  database system was not properly shut down; automatic recovery in progress
2019-10-09 01:33:56.347 UTC [21] LOG:  redo starts at 0/1633330
2019-10-09 01:33:56.353 UTC [21] LOG:  invalid record length at 0/16846A8: wanted 24, got 0
2019-10-09 01:33:56.353 UTC [21] LOG:  redo done at 0/1684640
2019-10-09 01:33:56.353 UTC [21] LOG:  last completed transaction was at log time 2019-10-09 01:33:53.442688+00
2019-10-09 01:33:56.372 UTC [1] LOG:  database system is ready to accept connections
2019-10-09 18:32:35.429 UTC [1] LOG:  received smart shutdown request
2019-10-09 18:32:35.448 UTC [1] LOG:  worker process: logical replication launcher (PID 27) exited with exit code 1
2019-10-09 18:32:36.630 UTC [14416] FATAL:  the database system is shutting down
running postgres /usr/local/bin/docker-entrypoint.sh postgres
2019-10-09 18:32:48.151 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2019-10-09 18:32:48.151 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2019-10-09 18:32:48.158 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2019-10-09 18:32:48.177 UTC [21] LOG:  database system was interrupted; last known up at 2019-10-09 01:38:58 UTC
2019-10-09 18:32:48.180 UTC [22] FATAL:  the database system is starting up
2019-10-09 18:32:48.239 UTC [21] LOG:  database system was not properly shut down; automatic recovery in progress
2019-10-09 18:32:48.242 UTC [21] LOG:  redo starts at 0/1688450
2019-10-09 18:32:48.242 UTC [21] LOG:  invalid record length at 0/1688530: wanted 24, got 0
2019-10-09 18:32:48.242 UTC [21] LOG:  redo done at 0/16884F8
2019-10-09 18:32:48.253 UTC [1] LOG:  database system is ready to accept connections
2019-10-09 18:35:50.657 UTC [65] LOG:  invalid length of startup packet
```

**Comment by omarroth** [comment]
The postgres container doesn't appear to be starting up correctly. How are you building the container?

**Comment by PipeItToDevNull** [comment]
I cloned the repo, and ran the docker file to build it. I tried to blow everything away and now get a `Caused by: No address found for postgres:5432 over TCP (Socket::Addrinfo::Error)` error

**Comment by PipeItToDevNull** [comment]
I properly pruned everything and fired it back up from source, same errors as the beginning (connection refused.)

**Comment by omarroth** [comment]
I'm assuming you're using:
```
$ docker-compose up
```
to build from docker?

**Comment by PipeItToDevNull** [comment]
`docker-compose up -d` to be specific

**Comment by omarroth** [comment]
Would you mind testing:

```
$ docker-compose down --rmi all -v --remove-orphans
$ docker-compose up
* cancel *
$ docker-compose up
```

**Comment by omarroth** [comment]
Should be fixed with 039cc30c07fa457850cd820e9e01b3afdf8920dc.

```
$ docker-compose build --pull
$ docker-compose up -d
```

**Comment by PipeItToDevNull** [comment]
Hello,

I blew up my environment, removed and re-puleld the repo down and now get connection refused from the DB. Not sure why, curl from invidious to posgres also gets connection refused so there is not even a TCP bind being accepted:

```
### first run - setting up invidious database
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data ... ok
creating subdirectories ... ok
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default timezone ... Etc/UTC
selecting dynamic shared memory implementation ... posix
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok

Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/data -l logfile start

waiting for server to start....2019-10-30 23:43:39.713 UTC [43] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2019-10-30 23:43:39.770 UTC [44] LOG:  database system was shut down at 2019-10-30 23:43:38 UTC
2019-10-30 23:43:39.776 UTC [43] LOG:  database system is ready to accept connections
 done
server started

/usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*

waiting for server to shut down....2019-10-30 23:43:39.799 UTC [43] LOG:  received fast shutdown request
2019-10-30 23:43:39.813 UTC [43] LOG:  aborting any active transactions
2019-10-30 23:43:39.818 UTC [43] LOG:  worker process: logical replication launcher (PID 50) exited with exit code 1
2019-10-30 23:43:39.818 UTC [45] LOG:  shutting down
2019-10-30 23:43:39.978 UTC [43] LOG:  database system is shut down
 done
server stopped

PostgreSQL init process complete; ready for start up.

/var/run/postgresql:5432 - accepting connections
CREATE ROLE
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE INDEX
CREATE TABLE
GRANT
CREATE TYPE
### invidious database setup finished
running postgres /usr/local/bin/docker-entrypoint.sh postgres
```
```
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Cannot establish connection (PQ::ConnectionError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Error connecting to 'postgres:5432': Connection refused (Errno)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: role "kemal" does not exist (PQ::PQError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: role "kemal" does not exist (PQ::PQError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Unhandled exception:  (DB::ConnectionRefused)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Cannot establish connection (PQ::ConnectionError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
Caused by: Error connecting to 'postgres:5432': Connection refused (Errno)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
```
I have no clue what is special about my environment, I may just blow this entire thing up and try it again on a debian machine
